syntax="proto3";

import "vtrc-errors.proto";
import "vtrc-rpc-lowlevel.proto";

//option optimize_for = LITE_RUNTIME;

package vtrc.rpc.auth;

// index 51000

enum hash_variant {
    HASH_NONE       =  0;

    // ---- crc ----  //
    HASH_CRC_16     =  1;
    HASH_CRC_32     =  2;
    HASH_CRC_64     =  3;

    // ---- sha ----  //
    HASH_SHA2_256   = 10;

};

enum transformer_variant {
    TRANSFORM_NONE       = 0;
    TRANSFORM_ERSEEFOR   = 1;
};

message init_protocol {
    repeated uint32 hash_supported             =  1; // @hash_variant
    repeated uint32 transform_supported        =  2; // @transformer_variant
}

message client_selection {
    uint32     hash                   =  1; // CRC_32
    uint32     transform              =  2; // default TRANSFORM_NONE
    bytes      id                     =  3;
}

message transformer_setup {
    bytes  salt1                      =  1;
    bytes  salt2                      =  2;
}

message session_setup {
    vtrc.rpc.session_options options = 1;
}

message init_capsule {
    string                       text  =  1;
    bool                         ready =  2;
    vtrc.rpc.errors.container    error =  3;

    bytes                        body = 10;
}
